src/aira.js

```diff
@@ -1,262 +1,273 @@
 var DEBUG = 0;
+var Aira;
 
-function Aira(impulse_response_calculator, var_model, view_model) {
-  this.impulse_response_calculator = impulse_response_calculator;
-  this.var_model = var_model;
-  this.view_model = view_model;
-}
-
-/**
- * Determines the node which has the most positive effect of all nodes in the network.
- * It does so by running the irf calculation for each node in the network, and summing
- * the irf responses of the other variables in the model. The cumulative score is returned.
- *
- * @returns [] a list of hashes containing each variable and the cumulative effect it has.
- */
-Aira.prototype.determineBestNodeFromAll = function() {
-  var variable, irf, cumulative_name, name, max, max_var,
-    result = {},
-    cumulative = {},
-    effects = [];
-
-  // Loop through all variables, and determine the impulse response of each variable on the variable to improve.
-  for (variable = 0; variable < this.var_model.number_of_variables; variable++) {
-
-    // Set the names for the nodes in the var model
-    name = this.var_model.get_node_name(variable);
-    cumulative_name = name + '_cumulative';
-
-    var only_significant = this.view_model.get_chk_bootstrap(),
-        bootstrap_iterations = this.view_model.get_bootstrap_iterations(),
-        confidence = .95;
-    // Shock the current variable in the loop
-    if(only_significant){
-      irf = this.impulse_response_calculator.significantImpulseResponseCalculation(variable, 1, this.view_model.get_steps(),
-          bootstrap_iterations, confidence);
-    } else {
-      irf = transpose(this.impulse_response_calculator.runImpulseResponseCalculation(variable, 1, this.view_model.get_steps()));
+Aira = (function () {
+
+    function Aira(impulse_response_calculator, var_model, view_model) {
+        this.impulse_response_calculator = impulse_response_calculator;
```

Comment:
create a test that tests whether these variables are class or instance wide. I remember a bug in this.